src/daft-micropartition/src/micropartition.rs

```diff
@@ -537,10 +445,15 @@ impl MicroPartition {
                     row_groups,
                     scan_task.pushdowns.filters.clone(),
                     scan_task.partition_spec(),
-                    cfg.io_config.clone().map(Arc::new).unwrap_or_default(),
+                    scan_task
+                        .storage_config
+                        .io_config
+                        .clone()
+                        .map(Arc::new)
+                        .unwrap_or_default(),
```

Comment:
Yeah seems like this is a common issue with needing to Arc the IOConfig on StorageConfig... Maybe we can fix it in a follow-on, or if simple enough just fix it here.